tests/internal/test_module.py

```diff
@@ -502,3 +502,14 @@ def test_module_watchdog_importlib_resources_files():
     import importlib.resources as r
 
     assert isinstance(r.files("namespace_test"), MultiplexedPath)
+
+
+@pytest.mark.subprocess
+def test_module_watchdog_does_not_rewrap():
+    """Ensures that self.loader.get_code() does not get re-wrapped (leading to a deeper stack)"""
+    from importlib import reload
+
+    from ddtrace import internal
+
+    for _ in range(10000):
```

Comment:
the idea here is to force a stack depth limit reached exception?

are there any other ways we can make sure we aren't double wrapping? like check that the loader's get_code isn't a wrapper of a wrapper?

Reply:
This actually mimics the issue we ran into, and I'm paranoid about `reload()` causing us to run into real-world problem, so it's kind of a twofer for me. It's a bit more integration-testy then unit-testy, I guess.

Reply:
@brettlangdon , added a separate test specifically to check we weren't changing the wrapper.